File: SampleApp/Objective-C/SampleApp/MoPub/VideoInterstitial/MoPubVideoInterstitialViewController.m
//
//  MoPubVideoInterstitialViewController.m
//  SampleApp
//
//  Copyright © 2020 fluct, inc. All rights reserved.
//

#import "MoPubVideoInterstitialViewController.h"
#import <MoPub/MoPub.h>

@interface MoPubVideoInterstitialViewController () <MPInterstitialAdControllerDelegate>
@property (nonatomic, weak) IBOutlet UIButton *showButton;
@property (nonatomic, nullable) MPInterstitialAdController *interstitial;
@end

// AdUnitIDを適切なものに変えてください
static NSString *const kAdUnitID = @"f9095a7d80e5405e84021cf54d5caf2a";

@implementation MoPubVideoInterstitialViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (IBAction)didTouchUpLoadAdWithButton:(UIButton *)button {
    self.interstitial = [MPInterstitialAdController interstitialAdControllerForAdUnitId:kAdUnitID];
    self.interstitial.delegate = self;
    [self.interstitial loadAd];
}

- (IBAction)didTouchUpShowAdWithButton:(UIButton *)button {
    if (self.interstitial.ready) {
        [self.interstitial showFromViewController:self];
    }
}

#pragma mark - MPInterstitialAdControllerDelegate

- (void)interstitialDidLoadAd:(MPInterstitialAdController *)interstitial {
    NSLog(@"%s", __FUNCTION__);
    self.showButton.enabled = YES;
}

- (void)interstitialDidFailToLoadAd:(MPInterstitialAdController *)interstitial withError:(NSError *)error {
    NSLog(@"%s, %@", __FUNCTION__, error);
}

- (void)interstitialWillAppear:(MPInterstitialAdController *)interstitial {
    NSLog(@"%s", __FUNCTION__);
}

- (void)interstitialDidAppear:(MPInterstitialAdController *)interstitial {
    NSLog(@"%s", __FUNCTION__);
}

- (void)interstitialDidExpire:(MPInterstitialAdController *)interstitial {
    NSLog(@"%s", __FUNCTION__);
}

- (void)interstitialWillDisappear:(MPInterstitialAdController *)interstitial {
    NSLog(@"%s", __FUNCTION__);
}

- (void)interstitialDidDisappear:(MPInterstitialAdController *)interstitial {
    NSLog(@"%s", __FUNCTION__);
    self.showButton.enabled = NO;
}

- (void)interstitialDidReceiveTapEvent:(MPInterstitialAdController *)interstitial {
    NSLog(@"%s", __FUNCTION__);
}

@end

File: SampleApp/Swift/SDK/SampleApp/MoPub/VideoInterstitial/MoPubVideoInterstitialViewController.swift
//
//  MoPubVideoInterstitialViewController.swift
//  SampleApp
//
//  Copyright © 2020 fluct, Inc. All rights reserved.
//

import UIKit
import MoPub

// AdUnitIDを適切なものに変えてください
private let adUnitID: String = "f9095a7d80e5405e84021cf54d5caf2a"

class MoPubVideoInterstitialViewController: UIViewController {

    @IBOutlet weak var showButton: UIButton!

    private var interstitial: MPInterstitialAdController?

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }

    @IBAction func didTouchUpLoadAd(button: UIButton) {
        self.interstitial = MPInterstitialAdController(forAdUnitId: adUnitID)
        self.interstitial?.delegate = self
        self.interstitial?.loadAd()
    }

    @IBAction func didTouchUpShowAd(button: UIButton) {
        guard let interstitial = self.interstitial, interstitial.ready else {
            return
        }
        interstitial.show(from: self)
    }

}

extension MoPubVideoInterstitialViewController: MPInterstitialAdControllerDelegate {

    func interstitialDidLoadAd(_ interstitial: MPInterstitialAdController!) {
        print(#function)
        showButton.isEnabled = true
    }

    func interstitialDidFail(toLoadAd interstitial: MPInterstitialAdController!, withError error: Error!) {
        print(#function, error!)
    }

    func interstitialWillAppear(_ interstitial: MPInterstitialAdController!) {
        print(#function)
    }

    func interstitialDidAppear(_ interstitial: MPInterstitialAdController!) {
        print(#function)
    }

    func interstitialDidExpire(_ interstitial: MPInterstitialAdController!) {
        print(#function)
    }

    func interstitialWillDisappear(_ interstitial: MPInterstitialAdController!) {
        print(#function)
    }

    func interstitialDidDisappear(_ interstitial: MPInterstitialAdController!) {
        print(#function)
        showButton.isEnabled = false
    }

    func interstitialDidReceiveTapEvent(_ interstitial: MPInterstitialAdController!) {
        print(#function)
    }

}
